airflow/models.py

```diff
@@ -2663,6 +2663,21 @@ def normalize_schedule(self, dttm):
 
         return dttm
 
+    @provide_session
+    def get_last_dagrun(self, session=None):
+        """
+        Returns the last dag run for this dag, None if there was none.
+        Last dag run can be any type of run eg. scheduled or backfilled.
+        Overriden DagRuns are ignored
+        """
+        DR = DagRun
+        last = session.query(DR).filter(
+            DR.dag_id == self.dag_id,
+            DR.external_trigger == False
```

Comment:
Won't do that as we dont use it anywhere. It is purely cosmetic as sqlalchemy supports this just fine.